test/realm_test.dart

```diff
@@ -853,6 +1019,57 @@ void openEncryptedRealm(List<int>? encryptionKey, List<int>? decryptionKey, {voi
   }
 }
 
+extension _FutureRealm on Future<Realm> {
+  Future<bool> thenIsCancelled() async {
+    return await then((value) {
+      expect(value, isNotNull);
+      expect(value.isClosed, false);
+      print("REALM CHECK: Realm returned before cancellation.");
+      return false;
+    }).onError(
+      (error, stackTrace) {
+        print("REALM CHECK: Cancelled before to return the Realm.");
+        return true;
+      },
+      test: (error) => error is CancelledException,
+    ).catchError((Object error) {
+      print("REALM CHECK: Unhandled exception.");
+      throw error;
+    }, test: (error) => error is! CancelledException);
+  }
+}
```

Comment:
```suggestion
extension _FutureRealm on Future<Realm> {
  Future<bool> thenIsCancelled() async {
    try {
      final value = await this;
      expect(value, isNotNull);
      expect(value.isClosed, false);
      print("REALM CHECK: Realm returned before cancellation.");
      return false;
    } on CancelledException {
      print("REALM CHECK: Cancelled before to return the Realm.");
      return true;
    } catch (_) {
      print("REALM CHECK: Unhandled exception.");
      rethrow;
    }
  }
}
```
Perhaps read up on https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures

Also, do we need the print statements?

Reply:
The ting is that I want just to attach `.then` and `.onError` here without to wait for the execution to complete. The waiting is managed in the tests accorting to the case. This extension only attach the checks that to be done when the execution completes. That's why its name is `.thenIsCancelled`. It is similar to` .then` but with a check for cancellation.

Reply:
I'm okay with the name, but I suggest we follow best practices from [effective dart](https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures):
> PREFER async/await over using raw futures.
Asynchronous code is notoriously hard to read and debug, even when using a nice abstraction like futures. The async/await syntax improves readability and lets you use all of the Dart control flow structures within your async code.